### ISSUE TITLE
Make external package and internal components folder as a separate group

### ISSUE THREAD

**Issue by szulcus** [import-js/eslint-plugin-import#2545]
I have this eslint configuration:
```json
    "import/order": [
      "error",
      {
        "pathGroups": [
          {
            "pattern": "@frontend/components",
            "group": "internal",
            "position": "after"
          },
          {
            "pattern": "~/components/**",
            "group": "internal",
            "position": "after"
          }
        ],
        "groups": [
          "builtin",
          "external",
          "internal",
          "unknown",
          "parent",
          "sibling",
          "index",
          "object",
          "type"
        ]
      }
    ]
```
I would like all my imports of `@frontend/components` and ` ~/components/ **` to be treated as a separate group. How can I do this?

**Comment by szulcus** [comment]
> Can you be more specific about what code you’d like, that this config considers incorrect?
Of course. I want to have all the imports sorted by default, except for components which will always have a line before an empty one, and will be in `~/components/** ` or `@frontend/components`. For example:
![Zrzut ekranu 2022-09-07 181826](https://user-images.githubusercontent.com/48328739/188929136-5323400e-72ef-409e-b538-6860ab88c9c3.png)

**Comment by ljharb** [comment]
How come your pathGroups don't account for `~/**`, since those are all internal?

Regardling newlines, you may also be interested in the `pathGroups.distinctGroup` option added in #2395, that will be in the next release.

**Comment by szulcus** [comment]
> How come your pathGroups don't account for `~/**`, since those are all internal?
> 
> Regardling newlines, you may also be interested in the `pathGroups.distinctGroup` option added in #2395, that will be in the next release.

I think I do not understand. Is it about something like that?
![image](https://user-images.githubusercontent.com/48328739/188935779-25277d7f-f02f-4479-8cff-3ae39c787d09.png)
If so, it still doesn't work 😅

**Comment by ljharb** [comment]
If you can provide a PR with a failing test case, I can likely solve it more quickly.

**Comment by szulcus** [comment]
Unfortunately the project and package are reserved, but I don't see any reason to reproduce it. I just want to have all the components separate. In Vue 3, in the `<script setup>` tag, there is no need to define the components, so I want to have them all in one place.

**Comment by ljharb** [comment]
@szulcus the reason is that without a reproduction case or regression tests, it can never be fixed - especially since I don't use Vue myself, so I'd have no idea how to set up a test case.

**Comment by szulcus** [comment]
> @szulcus the reason is that without a reproduction case or regression tests, it can never be fixed - especially since I don't use Vue myself, so I'd have no idea how to set up a test case.

I still think it's a pattern bug and doesn't require reproduction, but if you really need it, I created it for you:
[Nuxt Bridge ESLint import order reproduction.zip](https://github.com/import-js/eslint-plugin-import/files/9559343/Nuxt.Bridge.ESLint.import.order.reproduction.zip)

```console
yarn
yarn dev
```

File path: `~/pages/index.vue`
Environment: Visual Studio Code + Volar extension

![image](https://user-images.githubusercontent.com/48328739/189967607-f2ea58c7-62df-477e-9644-7b2e9302f16d.png)
